src/install.js

```
@@ -38,27 +51,53 @@ function getPlaywrightBrowserPath(kind) {
   return null;
 }
 
-async function installReplayBrowser(name, srcName, dstName) {
+// Installs a browser if it isn't already installed.
+async function installReplayBrowser(name, subdir, srcName, dstName) {
   const replayDir = getDirectory();
-  const playwrightDir = path.join(replayDir, "playwright");
+  const browserDir = path.join(replayDir, subdir);
 
-  if (fs.existsSync(path.join(playwrightDir, dstName))) {
+  if (fs.existsSync(path.join(browserDir, dstName))) {
     return;
   }
 
   const contents = await downloadReplayFile(name);
 
-  for (const dir of [replayDir, playwrightDir]) {
+  for (const dir of [replayDir, browserDir]) {
     if (!fs.existsSync(dir)) {
       fs.mkdirSync(dir);
     }
   }
-  fs.writeFileSync(path.join(playwrightDir, name), contents);
-  spawnSync("tar", ["xf", name], { cwd: playwrightDir });
-  fs.unlinkSync(path.join(playwrightDir, name));
+  fs.writeFileSync(path.join(browserDir, name), contents);
+  spawnSync("tar", ["xf", name], { cwd: browserDir });
+  fs.unlinkSync(path.join(browserDir, name));
 
   if (srcName != dstName) {
-    fs.renameSync(path.join(playwrightDir, srcName), path.join(playwrightDir, dstName));
+    fs.renameSync(path.join(browserDir, srcName), path.join(browserDir, dstName));
+  }
+}
+
+// Updates a browser if it is already installed.
+async function updateReplayBrowser(name, subdir, srcName, dstName, opts) {
+  const replayDir = getDirectory(opts);
+  const browserDir = path.join(replayDir, subdir);
+  const dstDir = path.join(browserDir, dstName);
+
+  if (fs.existsSync(dstDir)) {
+    // Remove the browser so installReplayBrowser will reinstall it. We don't have a way
+    // to see that the current browser is up to date.
+    fs.rmSync(dstDir, { force: true, recursive: true });
+  } else {
+    return;
```

Let's report an error here rather than stopping silently.

---

This isn't considered an error state --- replay-recordings update-browsers only updates browsers which are already installed, and doesn't download ones which haven't been installed by any packages on the system.  I figured users can use whatever packages they're using (playwright, puppeteer, cypress, more in the future) to download whichever replay browsers they need, and then can use replay-recordings update-browsers to update those without having to use any awkward workarounds.

---

Seems like if the user passed an invalid directory (e.g. by mistake) for update, we'd do nothing and the user wouldn't know. Am I reading the code incorrectly?

---

You're right, maybe we should count the number of browsers updated and exit with an error if none were found?